### PR TITLE
feat: pallet/asset: implement status_change() call

### LINKED ISSUE #267: pallet_asset: implement revoke() call
refer to revoke call in statement, and bring similar functionality in asset.

Expectation: revoke an asset or an instance (if its already active, and owner is matching).

### PR DESCRIPTION
**Pull request by amarts**
Fixes #267

### CODE REVIEW

File: pallets/asset/src/lib.rs [hunk 1]
```diff
@@ -383,6 +386,59 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		#[pallet::call_index(3)]
+		#[pallet::weight({0})]
+		pub fn revoke(
```
Comment (smohan-dw) [inline pallets/asset/src/lib.rs]: Revoke in asset context is a status change according to the AssetStatusOf enum. If the asset status is inactive or expired, no more issuance should happen. We need to check the expected behaviour of transfers if the underlying asset is not active.

File: pallets/asset/src/lib.rs [hunk 2]
```diff
@@ -383,6 +386,59 @@ pub mod pallet {
...
+			origin: OriginFor<T>,
+			asset_id: AssetIdOf,
+			instance_id: Option<AssetInstanceIdOf>,
```
Comment (smohan-dw) [inline pallets/asset/src/lib.rs:401]: We need to implement the status change behaviour for assets and instances separately.

**Comment by amarts** [comment]
As per one of the comment, instead of focusing on individual status of the asset/instance, one can keep `status_change()` itself as the key call, which can handle both revoke, restore. 

fixes: #267, #268

**Comment by amarts** [comment]
@smohan-dw does this makes sense now?

@adi-a11y the companion PR in cord.js would change too.

File: pallets/asset/src/lib.rs [hunk 3]
```diff
@@ -166,6 +166,13 @@ pub mod pallet {
 			from: AssetCreatorOf<T>,
 			to: AssetCreatorOf<T>,
 		},
+		/// An asset entry has been revoked.
```
Comment (amarts) [inline pallets/asset/src/lib.rs]: change the comment to reflect the new changes

**Comment by amarts** [comment]
@smohan-dw as per the discussion, for now, only 3 states are good, and this `status_change()` call handles asset status change or instance status change depending on the argument. 

Good to go IMO. 

I know this needs a version change in runtime, when deployed. But I am not sure if that should be with this PR, or a separate PR before a release (mentioning all changes in one shot).

**Review by smohan-dw — DISMISSED** [review]

File: pallets/asset/src/lib.rs (hunk 2 above)
Comment (amarts) [reply pallets/asset/src/lib.rs:401]: @smohan-dw looks like the comment is bit older than the review. Currently, the status change takes instance status change if instance is sent. Asset state is not changed. If instance is None, then asset status change happens.

File: pallets/asset/src/lib.rs (hunk 1 above)
Comment (amarts) [reply pallets/asset/src/lib.rs]: The logic of transfer and issuance only checks for 'ACTIVE' already in code, so any other status change would result in stopping of transfer and issuance from happening.

**Comment by amarts** [comment]
Merging this as @smohan-dw's approval was present for commit without the version change. With this, we will be comfortable to update the existing chain without any major issues.

**Pull request merged**